src/main/kotlin/io/zerobase/smarttracing/features/devices/DevicesResource.kt

```diff
@@ -63,4 +80,37 @@ class DevicesResource(val dao: DevicesDao) {
     ) {
         dao.updateCheckInLocation(DeviceId(deviceId), ScanId(checkInId), loc)
     }
+
+    @Path("/{id}/reports/tests")
+    @POST
+    @Creator
+    fun selfReportTestResult(@PathParam("id") id: String, report: SelfReportedTestResult): IdWrapper {
+        val deviceId = DeviceId(id)
+        return dao.recordTestResult(TestResult(
+            reportedBy = deviceId,
+            testedParty = deviceId,
+            result = report.result,
+            testDate = report.testDate,
+            verified = false,
+            timestamp = report.timestamp
+        )).let { IdWrapper(it) }
```

Comment:
it looks like both recordTestResult and recordSymptoms both return a ReportId. What is the difference between` .let{ IdWrapper(it) } `and `.let(::IdWrapper)?`

Reply:
one is a lambda and one is a method reference. i fixed it in the other PR to use the reference syntax instead of the lambda. i need to rebase this PR on that one so it picks up the changes